packages/uni_ui/lib/restaurant_card.dart

```diff
@@ -0,0 +1,136 @@
+import 'package:flutter/material.dart';
+import 'package:uni_ui/generic_card.dart';
+import 'package:phosphor_flutter/phosphor_flutter.dart';
+
+class MenuItem {
+  final String name;
+  final String type;
+
+  MenuItem({required this.name, required this.type});
+}
+
+class RestaurantCard extends StatefulWidget {
+  final String title;
+  final String establishmentType;
+  final List<MenuItem> menuItems;
+
+  const RestaurantCard({
+    super.key,
+    required this.title,
+    required this.establishmentType,
+    required this.menuItems,
+  });
+
+  @override
+  _RestaurantCardState createState() => _RestaurantCardState();
+}
+
+class _RestaurantCardState extends State<RestaurantCard> {
+  bool isFavorite = false;
+
+  void _toggleFavorite() {
+    setState(() {
+      isFavorite = !isFavorite;
+    });
+  }
+
+  IconData? _getIconForMenuItemType(String type) {
+    switch (type.toLowerCase()) {
+      case 'meat':
+        return PhosphorIconsBold.cow;
+      case 'fish':
+        return PhosphorIconsBold.fish;
+      default:
+        return PhosphorIconsBold.leaf;
+    }
+  }
+
+  Color _getIconColorForMenuItemType(String type) {
+    return Theme.of(context).colorScheme.primary;
+  }
```

Comment:
Looks a little redundant. I would remove that and put the theme property directly on line 116.